File: frontend/lib/model/joblisting/command/JoinJoblistingCommand.ts
export default class JoinJoblistingCommand {
    private readonly name: string;
    private readonly owner: string;
    private readonly project: string;

    constructor(name: string, owner: string, project: string) {
        this.name = name;
        this.owner = owner;
        this.project = project;
    }
}

File: backend/src/main/java/de/garrafao/phitag/application/joblisting/data/command/JoinJoblistingCommand.java
package de.garrafao.phitag.application.joblisting.data.command;

import org.apache.commons.lang3.Validate;

import lombok.Getter;

@Getter
public class JoinJoblistingCommand {
    
    private final String name;
    private final String owner;
    private final String project;

    public JoinJoblistingCommand(final String name, final String owner, final String project) {
        Validate.notBlank(name, "name must not be blank");
        Validate.notBlank(owner, "owner must not be blank");
        Validate.notBlank(project, "project must not be blank");
        
        this.name = name;
        this.owner = owner;
        this.project = project;
    }
}
